src/MySqlConnector/Utilities/ActivitySourceHelper.cs

```diff
@@ -36,7 +36,13 @@ internal static class ActivitySourceHelper
 		return activity;
 	}
 
-	public static void SetSuccess(this Activity activity) => activity.SetTag(StatusCodeTagName, "OK");
+	public static void SetSuccess(this Activity activity)
+	{
+		if (activity.Duration == TimeSpan.Zero)
```

Comment:
What's the purpose of this change?

Reply:
Oh, is this a workaround for a lack of an `IsStopped` property (https://github.com/dotnet/runtime/issues/63353) to check if the Activity is still running? A comment would be helpful here, but I think checking `Status` would be clearer.

Reply:
uh, activity only supports `Status` property in .net 6 and above.

Here is the resolve to set the Exception first, and then set the Success to cause the state to be overwritten.

set the Exception should have the highest priority.

Reply:
Thanks; I hadn't checked the docs thoroughly to see where the API was available.